libraries/exoplayer_rtsp/src/main/java/androidx/media3/exoplayer/rtsp/reader/RtpVP8Reader.java

```diff
@@ -0,0 +1,217 @@
+import static androidx.media3.common.util.Util.castNonNull;
+
+import androidx.media3.common.C;
+import androidx.media3.common.Format;
+import androidx.media3.common.ParserException;
+import androidx.media3.common.util.Log;
+import androidx.media3.common.util.ParsableByteArray;
+import androidx.media3.common.util.Util;
+import androidx.media3.exoplayer.rtsp.RtpPacket;
+import androidx.media3.exoplayer.rtsp.RtpPayloadFormat;
+import androidx.media3.extractor.ExtractorOutput;
+import androidx.media3.extractor.TrackOutput;
+import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
+
+/**
+ * Parses a VP8 byte stream carried on RTP packets, and extracts VP8 individual video frames as
+ * defined in RFC7741.
+ */
+/* package */ final class RtpVP8Reader implements RtpPayloadReader {
+  private static final String TAG = "RtpVP8Reader";
+
+  private static final long MEDIA_CLOCK_FREQUENCY = 90_000;
+
+  private final RtpPayloadFormat payloadFormat;
+
+  private @MonotonicNonNull TrackOutput trackOutput;
+  @C.BufferFlags private int bufferFlags;
+
+  private long firstReceivedTimestamp;
+  private int previousSequenceNumber;
+  /** The combined size of a sample that is fragmented into multiple RTP packets. */
+  private int fragmentedSampleSizeBytes;
+  private long startTimeOffsetUs;
+  private boolean gotFirstPacketOfVP8Frame;
+  private boolean isKeyFrame;
+  private boolean isOutputFormatSet;
+
+  /** Creates an instance. */
+  public RtpVP8Reader(RtpPayloadFormat payloadFormat) {
+    this.payloadFormat = payloadFormat;
+    firstReceivedTimestamp = C.TIME_UNSET;
+    previousSequenceNumber = C.INDEX_UNSET;
+    fragmentedSampleSizeBytes = 0;
+    gotFirstPacketOfVP8Frame = false;
+    isKeyFrame = false;
+    isOutputFormatSet = false;
+  }
+
+  @Override
+  public void createTracks(ExtractorOutput extractorOutput, int trackId) {
+    trackOutput = extractorOutput.track(trackId, C.TRACK_TYPE_VIDEO);
+    castNonNull(trackOutput).format(payloadFormat.format);
+  }
+
+  @Override
+  public void onReceivingFirstPacket(long timestamp, int sequenceNumber) {}
+
+  @Override
+  public void consume(ParsableByteArray data, long timestamp, int sequenceNumber, boolean rtpMarker)
+      throws ParserException {
+    checkStateNotNull(trackOutput);
+
+    if (parseVP8Descriptor(data, sequenceNumber)) {
+      //  VP8 Payload Header, RFC7741 Section 4.3
+      //  0 1 2 3 4 5 6 7
+      //  +-+-+-+-+-+-+-+-+
+      //  |Size0|H| VER |P|
+      //  +-+-+-+-+-+-+-+-+
+      // P: Inverse key frame flag.
+      if (fragmentedSampleSizeBytes == 0 && gotFirstPacketOfVP8Frame) {
+        isKeyFrame = (data.peekUnsignedByte() & 0x01) == 0;
+      }
+      if (!isOutputFormatSet) {
+        // Parsing frame data to get width and height, RFC6386 Section 9.1
+        int currPosition = data.getPosition();
+        data.setPosition(currPosition + 6);
+        int width = data.readLittleEndianUnsignedShort() & 0x3fff;
```

Comment:
Do we need littleEndian here? I assume the data here is in "network order". Also this library could be used on big-endian devices, will using littleEndian method here limit the usecases?

Reply:
According to this width and height are save in littleEndianess.